core/src/main/java/com/linecorp/armeria/common/logging/PrometheusMetricRequestDecorator.java

```diff
@@ -0,0 +1,299 @@
+        }
+
+        private Summary.Builder addQuantiles(Summary.Builder builder) {
+            return builder.quantile(.5, .01)
+                          .quantile(.75, .01)
+                          .quantile(.95, .01)
+                          .quantile(.98, .01)
+                          .quantile(.99, .01)
+                          .quantile(.999, .01);
+        }
+
+        private <T extends Collector> T create(String name,
+                                               PrometheusRegistryWrapper collectorRegistry,
+                                               Function<String, T> ifAbsent) {
+            return collectorRegistry.create(name, ifAbsent);
+        }
+
+        public MetricAdapter time(String[] values, long time) throws Exception {
+            timer.labels(values).observe(time);
+            return this;
+        }
+
+        public MetricAdapter success(String[] values) {
+            success.labels(values).inc();
+            return this;
+        }
+
+        public MetricAdapter failure(String[] values) {
+            failure.labels(values).inc();
+            return this;
+        }
+
+        public MetricAdapter incActiveRequests(String[] values) {
+            activeRequests.labels(values).inc();
+            return this;
+        }
+
+        public MetricAdapter decActiveRequests(String[] values) {
+            activeRequests.labels(values).dec();
+            return this;
+        }
+
+        public MetricAdapter requestBytes(String[] values, RequestLog log) {
+            requestBytes.labels(values).observe(log.requestLength());
+            return this;
+        }
+
+        public MetricAdapter responseBytes(String[] values, RequestLog log) {
+            responseBytes.labels(values).observe(log.responseLength());
+            return this;
+        }
+    }
+
+    private final String helpString = "Armeria client/server request metric";
+    private final Function<RequestLog, Map<? extends MetricLabel, String>> labelingFunction;
+    private final MetricAdapter metricAdapter;
+
+
+    /**
+     * Creates a new instance that decorates the specified {@link Service}.
+     */
+    private PrometheusMetricRequestDecorator(
+            PrometheusRegistryWrapper collectorRegistry,
+            MetricLabel[] metricLabels,
+            Function<RequestLog, Map<? extends MetricLabel, String>> labelingFunction,
+            String prefix) {
+        this.labelingFunction = labelingFunction;
+        this.metricAdapter = new MetricAdapter(collectorRegistry, metricLabels, prefix, helpString);
+    }
+
+    @FunctionalInterface
+    private interface ThrowsBiFunction<C, I, O> {
+        O apply(C c, I i) throws Exception;
+    }
+
+    private <C extends RequestContext> O request(ThrowsBiFunction<C, I, O> function, C ctx, I req)
+            throws Exception {
+        ctx.log().addListener(
+                log -> {
+                    final String[] values = getLabelValues(log);
```

Comment:
That can work. Basically I copied what we've been using at https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/server/logging/DropwizardMetricCollectingService.java#L94
https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/server/logging/DropwizardMetricCollectingService.java#L137

I was wondering what the motivation was  myself